s3_file_uploads/views.py

```diff
@@ -14,14 +19,17 @@ def get_queryset(self):
         return UploadedFile.objects.filter(user=self.request.user)
 
     def create(self, request, *args, **kwargs):
-        serializer_data = {**request.data, 'user': request.user.id}
-        serializer = self.get_serializer(data=serializer_data)
-        serializer.is_valid(raise_exception=True)
-        self.perform_create(serializer)
-        headers = self.get_success_headers(serializer.data)
-        uploaded_file = serializer.instance
+        file_serializer = self.get_serializer(
+            data={**get_file_data_from_request_data(request.data), 'user': request.user.id}
+        )
+        file_serializer.is_valid(raise_exception=True)
+        self.perform_create(file_serializer)
+        headers = self.get_success_headers(file_serializer.data)
+        uploaded_file = file_serializer.instance
         uploaded_file_data = UploadedFileSerializer(instance=uploaded_file).data
-        uploaded_file_data['upload_form'] = uploaded_file.get_upload_form()
+        uploaded_file_data['upload_form'] = uploaded_file.get_upload_form(
+            acl_type=validate_acl_type(get_acl_data_from_request_data(request.data))
```

Comment:
Would it be better to add this validation to the initial serializer? rather than after creating the file object etc

Reply:
Serailiser is much better